other/upgrade.php

```diff
@@ -4997,49 +5010,43 @@ function MySQLConvertOldIp($targetTable, $oldCol, $newCol, $limit = 50000, $setS
 		$request = $smcFunc['db_query']('', '
 			SELECT DISTINCT {raw:old_col}
 			FROM {db_prefix}{raw:table_name}
-			WHERE {raw:new_col} IS NULL AND
-				{raw:old_col} != {string:unknown} AND
-				{raw:old_col} != {string:empty}
+			WHERE {raw:new_col} = {string:empty}
```

Comment:
Shouldn't this check for `NULL` instead of empty string so that it can skip those on subsequent future runs?

Reply:
No, it should check for empty string.  That is how the field is initialized.

Note that when you pass anything other than an IP to $smcFunc for an inet datatype, is stores NULL.  So we need to differentiate between 'not yet done' and 'done but funky'.  empty = not yet done.  NULL = funky.